File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLORS = %w(black brown red orange yellow green blue violet grey white)

  attr_reader :value

  def self.value(colors)
    new(colors).value
  end

  def initialize(colors)
    @value = colors
      .map(&method(:color_value))
      .join
      .to_i
  end

  private

  def color_value(color)
    COLORS.index(color) or raise ArgumentError, "Bad color: #{color}"
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
export const value = (colors) => {
  return parseInt(
    colors.reduce(
      (acc, color) => acc + COLORS.indexOf(color),
      ""
    )
  );
};

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];
